src/pudl/analysis/classify_plants_ferc1.py

```diff
@@ -679,7 +681,8 @@ def fuel_by_plant_ferc1(
                 "plant_name_ferc1",
                 "report_year",
                 "fuel_type_code_pudl",
-            ]
+            ],
+            observed=True,
```

Comment:
Are you sure we want `observed=True` here?  This column is pivoted such that there's a column per fuel type later on, and if there are missing values they'll become NA. Ah. Okay but it looks like we fill NA with 0.0 so I guess it's fine. As long as every fuel type is observed at least once.

Reply:
We were previously converting `fuel_type_code_pudl` into a `str` column before this because this column in our metadata is (correctly) a categorical. I suspected early on that the `str` type conversion was a part of the problem (it wasn't). But the only reason we needed that conversion was because of the semi-new `groupby` behavior regarding categoricals.

Reply:
the outcome of the new version of the table passes the `pd.testing.assert_frame_equal` so i feel pretty confident that the new behavior is the same.

Reply:
Okay thanks for showing me your debugging. I'm convinced this is a new pandas behavior somehow and now an upstream data issue. Seems very strange though.